src/mscorlib/src/System/Runtime/InteropServices/GcHandle.cs

```diff
@@ -89,70 +88,47 @@ public static GCHandle Alloc(Object value, GCHandleType type)
             return new GCHandle(value, type);
         }
 
-
         // Frees a GC handle.
         public void Free()
         {
-            // Copy the handle instance member to a local variable. This is required to prevent
-            // race conditions releasing the handle.
-            IntPtr handle = m_handle;
-
             // Free the handle if it hasn't already been freed.
-            if (handle != IntPtr.Zero && Interlocked.CompareExchange(ref m_handle, IntPtr.Zero, handle) == handle)
-            {
+            IntPtr handle = Interlocked.Exchange(ref m_handle, IntPtr.Zero);
```

Comment:
Doesn't need `CompareExchange` as `m_handle` always ends up as `IntPtr.Zero`; so can just `Exchange` it with `IntPtr.Zero` and the perform the tests on the result